coupler/atmos_ocean_fluxes.F90

```diff
@@ -118,9 +117,9 @@ function aof_set_coupler_flux(name, flux_type, implementation, atm_tr_index, par
     character(len=*), intent(in)                :: flux_type !< flux_type
     character(len=*), intent(in)                :: implementation !< implementation
     integer, intent(in), optional               :: atm_tr_index !< atm_tr_index
-    real, intent(in), dimension(:), optional    :: param !< param
+    real(r8_kind), intent(in), dimension(:), optional    :: param !< param
     logical, intent(in), dimension(:), optional :: flag !< flag
-    real, intent(in), optional                  :: mol_wt !< mol_wt
+    real(r8_kind), intent(in), optional                  :: mol_wt !< mol_wt
```

Comment:
Do you think it'd be better if mol_wt was a class(*) here?

Reply:
Here's where it ends up being used in this routine:

https://github.com/NOAA-GFDL/FMS/blob/5d8e7ffeb25622b4be39ce22c7d5756a1308c7b2/coupler/atmos_ocean_fluxes.F90#L298-L302

But then also again here:

https://github.com/NOAA-GFDL/FMS/blob/5d8e7ffeb25622b4be39ce22c7d5756a1308c7b2/coupler/atmos_ocean_fluxes.F90#L645

I think fm_util_get_real is r8_kind only right cause of the read in strings?

Reply:
that and because there was no way to tell what precision the user wanted based on the input arguments.  I'm wondering if the users will pass in r4_kind mol_wt here because they're assuming that the r4 version of bc will be populated

Reply:
Yeah to be safe I'll switch it over to a class(*), since it's intent(in) should be fine.